dbt_subprojects/dex/models/prices/dex_prices_block_solana.sql

```diff
@@ -0,0 +1,167 @@
+{{ config(
+    schema = 'dex'
+    , alias = 'prices_block_solana'
+    , partition_by = ['block_month']
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['block_month', 'blockchain', 'contract_address', 'symbol', 'decimals', 'block_number', 'block_time']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+)
+}}
+
+with dex_trades_raw as (
+    select
+        blockchain
+        , block_number
+        , block_time
+        , token_bought_address
+        , token_bought_amount_raw
+        , token_bought_amount
+        , token_sold_address
+        , token_sold_amount_raw
+        , token_sold_amount
+        , amount_usd
+        , array[token_bought_address, token_sold_address] as tokens_swapped
+    from
+        {{ ref('dex_solana_trades') }}
+    where
+        1 = 1
+        and amount_usd > 0
+        {% if is_incremental() %}
+        and {{ incremental_predicate('block_time') }}
+        {% endif %}
+),
+dex_trades as (
+    select distinct
+        t.blockchain
+        , t.block_number
+        , t.block_time
+        , t.token_bought_address
+        , t.token_bought_amount_raw
+        , t.token_bought_amount
+        , t.token_sold_address
+        , t.token_sold_amount_raw
+        , t.token_sold_amount
+        , t.amount_usd
+    from
+        dex_trades_raw as t
+    --only output swaps which contain a trusted token
+    inner join {{ source('prices', 'trusted_tokens') }} as tt
+        on t.blockchain = tt.blockchain
+        and contains(t.tokens_swapped, tt.contract_address)
+),
+dex_bought as (
+    select
+        d.blockchain
+        , d.token_bought_address as contract_address
+        , t.symbol as symbol
+        , t.decimals as decimals
+        , d.block_number as block_number
+        , d.block_time as block_time
+        , d.token_bought_amount as amount
+        , d.amount_usd
+        , coalesce(d.amount_usd/d.token_bought_amount, d.amount_usd/(d.token_bought_amount_raw/POW(10, t.decimals))) as price
+    from
+        dex_trades as d
+    inner join {{ source('tokens', 'erc20') }} as t
```

Comment:
not sure we have solana in this table

we will need to check in the solana lineage how it leverages decimals to calculate amount_usd and then do similar here

doubtful our automated source for erc20 has solana tokens